src/main/scala/devices/tilelink/CLINT.scala

```diff
@@ -12,108 +12,53 @@ import freechips.rocketchip.subsystem._
 import freechips.rocketchip.tilelink._
 import freechips.rocketchip.util._
 
-object CLINTConsts
-{
-  def msipOffset(hart: Int) = hart * msipBytes
-  def timecmpOffset(hart: Int) = 0x4000 + hart * timecmpBytes
-  def timeOffset = 0xbff8
-  def msipBytes = 4
-  def timecmpBytes = 8
-  def size = 0x10000
-  def timeWidth = 64
-  def ipiWidth = 32
-  def ints = 2
-}
-
-case class CLINTParams(baseAddress: BigInt = 0x02000000, intStages: Int = 0)
-{
-  def address = AddressSet(baseAddress, CLINTConsts.size-1)
-}
-
-case object CLINTKey extends Field[Option[CLINTParams]](None)
-
-case class CLINTAttachParams(
-  slaveWhere: TLBusWrapperLocation = CBUS
+// if isACLINT is false, the code will generate a clint
+case class CLINTParams(
+  isACLINT: Boolean = false,
+  mtimer: Option[MTIMERParams]  = Some(MTIMERParams()),
+  mswi: Option[MSWIParams]      = Some(MSWIParams())
 )
```

Comment:
Does `mtimer=None` or `mswi=None` make sense?

Reply:
Yes, let me display dts without clint later.

Reply:
In that case, CLINTKey should just be None. I'm not sure I understand why these sub-params need to be option typed too

Reply:
Because in the spec,  The mtimer may not exist or the mswi may not exist.

Reply:
Ok. Can you add a require to this class to check that both cannot be set to None?

Also, can you verify that all three cases generate properly?

Reply:
Yes, I will check three cases:
```
mtimer: not None, mswi: not None
mtimer: None, mswi: not None
mtimer: not None, mswi: None
```